lstchain/image/muon/intensity_fitter.py

```diff
@@ -0,0 +1,562 @@
+"""
+Class for performing a HESS style 2D fit of muon images
+
+To do:
+    - Deal with astropy untis better, currently stripped and no checks made
+    - unit tests
+    - create container class for output
+
+"""
+import numpy as np
+from scipy.ndimage.filters import correlate1d
+from iminuit import Minuit
+from astropy import units as u
+from scipy.constants import alpha
+from scipy.stats import norm
+from astropy.coordinates import SkyCoord
+from functools import lru_cache
+
+from ctapipe.containers import MuonEfficiencyContainer
+from ctapipe.coordinates import CameraFrame, TelescopeFrame
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import FloatTelescopeParameter, IntTelescopeParameter
+
+
+# ratio of the areas of the unit circle and a square of side lengths 2
+CIRCLE_SQUARE_AREA_RATIO = np.pi / 4
+
+
+def chord_length(radius, rho, phi):
+    """
+    Function for integrating the length of a chord across a circle
+
+    Parameters
+    ----------
+    radius: float
+        radius of circle
+    rho: float
+        fractional distance of impact point from circle center
+    phi: ndarray in radians
+        rotation angles to calculate length
+
+    Returns
+    -------
+    ndarray: chord length
```

Comment:
incomplete docstring here.
phi can also be a float and in that case, a float is returned